client/utils/auth.py

```diff
@@ -160,14 +164,17 @@ def refresh_local_token(server):
 def perform_oauth(code_fn, *args, **kwargs):
     try:
         access_token, expires_in, refresh_token = code_fn(*args, **kwargs)
+    except UnicodeDecodeError as e:
+        with format.block('-'):
+            print("Authentication error\n:{}".format(HOSTNAME_ERROR_MESSAGE))
     except OAuthException as e:
         with format.block('-'):
             print("Authentication error: {}".format(e.error.replace('_', ' ')))
             if e.error_description:
                 print(e.error_description)
-        return None
-    update_storage(access_token, expires_in, refresh_token)
-    return access_token
+    else:
```

Comment:
Used an `else` block to clean up the code here a bit.  In python, if the try block succeeds, the else block executes.